### ISSUE TITLE
Separate data/control planes in the token_id passing in GRPO

### ISSUE THREAD

**Issue by guyueh1** [NVIDIA-NeMo/RL#1496]
**Is your feature request related to a problem? Please describe.**
Right now in nemo-rl GRPO, the generation workers will return the token_ids to header node on cpu, and header node will perform sequence packing and pass the token_ids as input arguments to train() or logprob() functions of policy workers. This is inefficient when global batch size or mean sequence length are large, as the large size of token_ids may cause excessive memory spilling to disk in the header node. 

**Describe the solution you'd like**
Implement a mode that minimizes the transfer of token_ids between CPU and GPU, ideally always keep token_ids on GPU and use GPU direct transfer to pass them to workers; workers only return metadata to CPU for running sequence packing, loss function should be calculated on GPU.

**Describe alternatives you've considered**

**Additional context**

**Comment by guyueh1** [comment]
Tracked internally by @ZhiyuLi-Nvidia , closing this public ticket
